web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx

```diff
@@ -0,0 +1,297 @@
+import { App } from 'teleport/services/apps';
+
+type Props = {
+  resource: UnifiedResource;
+};
+
+export const ResourceActionButton = ({ resource }: Props) => {
+  switch (resource.kind) {
+    case 'node':
+      return <NodeConnect node={resource} />;
+    case 'app':
+      return <AppLaunch app={resource} />;
+    case 'db':
+      return <DatabaseConnect database={resource} />;
+    case 'kube_cluster':
+      return <KubeConnect kube={resource} />;
+    case 'windows_desktop':
+      return <DesktopConnect desktop={resource} />;
+    default:
+      return null;
+  }
+};
+
+const NodeConnect = ({ node }: { node: Node }) => {
+  const { clusterId } = useStickyClusterId();
+  const startSshSession = (login: string, serverId: string) => {
+    const url = cfg.getSshConnectRoute({
+      clusterId,
+      serverId,
+      login,
+    });
+
+    openNewTab(url);
+  };
+
+  function handleOnOpen() {
+    return makeNodeOptions(clusterId, node);
+  }
+
+  const handleOnSelect = (e: React.SyntheticEvent, login: string) => {
+    e.preventDefault();
+    return startSshSession(login, node.id);
+  };
+
+  return (
+    <MenuLogin
+      getLoginItems={handleOnOpen}
+      onSelect={handleOnSelect}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'right',
+      }}
+      anchorOrigin={{
+        vertical: 'center',
+        horizontal: 'right',
+      }}
+    />
+  );
+};
+
+const DesktopConnect = ({ desktop }: { desktop: Desktop }) => {
+  const { clusterId } = useStickyClusterId();
+  const startRemoteDesktopSession = (username: string, desktopName: string) => {
+    const url = cfg.getDesktopRoute({
+      clusterId,
+      desktopName,
+      username,
+    });
+
+    openNewTab(url);
+  };
+
+  function handleOnOpen() {
+    return makeDesktopLoginOptions(clusterId, desktop.name, desktop.logins);
+  }
+
+  function handleOnSelect(e: React.SyntheticEvent, login: string) {
+    e.preventDefault();
+    return startRemoteDesktopSession(login, desktop.name);
+  }
+
+  return (
+    <MenuLogin
+      getLoginItems={handleOnOpen}
+      onSelect={handleOnSelect}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'right',
+      }}
+      anchorOrigin={{
+        vertical: 'center',
+        horizontal: 'right',
+      }}
+    />
+  );
+};
+
+const AppLaunch = ({ app }: { app: App }) => {
+  const {
+    launchUrl,
+    awsConsole,
+    awsRoles,
+    fqdn,
+    clusterId,
+    publicAddr,
+    isCloudOrTcpEndpoint,
+    samlApp,
+    samlAppSsoUrl,
+  } = app;
+  let $btn;
+  if (awsConsole) {
+    $btn = (
+      <AwsLaunchButton
+        awsRoles={awsRoles}
+        fqdn={fqdn}
+        clusterId={clusterId}
+        publicAddr={publicAddr}
+      />
+    );
+  } else if (isCloudOrTcpEndpoint) {
+    $btn = (
+      <ButtonBorder
+        disabled
+        width="90px"
+        size="small"
+        title="Cloud or TCP applications cannot be launched by the browser"
+      >
+        Launch
```

Comment:
I see you changed the text to mixed case here, but the problem is that the buttons use uppercase by default. We first need to expose this as a button attribute (if it's not already exposed), and then we'll need to use this ability here.